BuildResidentialHPXML/measure.rb

```diff
@@ -3649,30 +3685,42 @@ def self.set_pv_systems(hpxml, runner, args, weather)
   end
 
   def self.set_lighting(hpxml, runner, args)
-    hpxml.lighting_groups.add(id: 'Lighting_TierI_Interior',
-                              location: HPXML::LocationInterior,
-                              fration_of_units_in_location: 0.5, # FIXME
-                              third_party_certification: HPXML::LightingTypeTierI)
-    hpxml.lighting_groups.add(id: 'Lighting_TierI_Exterior',
-                              location: HPXML::LocationExterior,
-                              fration_of_units_in_location: 0.5, # FIXME
-                              third_party_certification: HPXML::LightingTypeTierI)
-    hpxml.lighting_groups.add(id: 'Lighting_TierI_Garage',
-                              location: HPXML::LocationGarage,
-                              fration_of_units_in_location: 0.5, # FIXME
-                              third_party_certification: HPXML::LightingTypeTierI)
-    hpxml.lighting_groups.add(id: 'Lighting_TierII_Interior',
-                              location: HPXML::LocationInterior,
-                              fration_of_units_in_location: 0.25, # FIXME
-                              third_party_certification: HPXML::LightingTypeTierII)
-    hpxml.lighting_groups.add(id: 'Lighting_TierII_Exterior',
-                              location: HPXML::LocationExterior,
-                              fration_of_units_in_location: 0.25, # FIXME
-                              third_party_certification: HPXML::LightingTypeTierII)
-    hpxml.lighting_groups.add(id: 'Lighting_TierII_Garage',
-                              location: HPXML::LocationGarage,
-                              fration_of_units_in_location: 0.25, # FIXME
-                              third_party_certification: HPXML::LightingTypeTierII)
+    if args[:lighting_fraction_cfl_interior].is_initialized
+      hpxml.lighting_groups.add(id: 'Lighting_TierI_Interior',
+                                location: HPXML::LocationInterior,
+                                fration_of_units_in_location: args[:lighting_fraction_cfl_interior].get,
+                                third_party_certification: HPXML::LightingTypeTierI)
+    end
+    if args[:lighting_fraction_cfl_exterior].is_initialized
+      hpxml.lighting_groups.add(id: 'Lighting_TierI_Exterior',
+                                location: HPXML::LocationExterior,
+                                fration_of_units_in_location: args[:lighting_fraction_cfl_exterior].get,
+                                third_party_certification: HPXML::LightingTypeTierI)
+    end
```

Comment:
This provides too much flexibility as we need all 6 inputs provided or none provided. If just one input is provided, for example, that's not sufficient and will generate errors from EPvalidator. I suggest making all of the inputs required with default values.